File: projects/spring-boot-restv2/demo/src/main/java/com/sourabh/demo/iservice/IRoomService.java
package com.sourabh.demo.iservice;

import com.sourabh.demo.model.Room;

import java.util.List;

public interface IRoomService {
    public Room findById(Integer id);
    public Room findByIdAvailable(Integer id);
    public Room findByType(String type);
    public Room findByTypeAvailable(String type);
    public List<Room> getAll();
    public List<Room> getAllAvailable();
    public Room addOrUpdate(Room room);
    public List<Room> addAll(List<Room> rooms);
    public Room removeRoom(int id);
    public void updateRoomCount(Room room,int count);
}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/iservice/IRoomService.kt
package com.sourabh.hotelreservation.iservice

import com.sourabh.hotelreservation.model.Room

interface IRoomService {
    fun findById(id:Int): Room
    fun findByIdAvailable(id:Int):Room
    fun findByType(type:String):Room
    fun findByTypeAvailable(type:String):Room
    fun getAll():MutableList<Room>?
    fun getAllAvailable(): MutableList<Room>?
    fun addOrUpdate(room:Room):Room
    fun addAll(rooms:MutableList<Room>):MutableList<Room>
    fun removeRoom(id:Int):Room
    fun updateRoomCount(room:Room,count:Int)

}